### PR TITLE
[NetworkManager] Get system-connections files from other locations

### PR DESCRIPTION
**Pull request by jcastill**
system-connections files can be found not only under /etc/NetworkManager, but also /usr/lib/NetworkManager and /run/NetworkManager, so we need to capture and post process these locations as well.

Resolves: #3875

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [X] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [X] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

File: sos/report/plugins/networkmanager.py [hunk 1]
```diff
@@ -110,16 +117,19 @@ def test_nm_status(version=1):
             "nmcli dev show .*": "nmcli_dev_show_sos"
         })
 
-    def postproc(self):
-        for _, _, files in os.walk(
-                "/etc/NetworkManager/system-connections"):
+    def postproc_system_connections(self, sc_path):
+        for _, _, files in os.walk(sc_path):
```
Comment (champtar) [inline sos/report/plugins/networkmanager.py:121]: Why not use `do_path_regex_sub()` ?
https://github.com/sosreport/sos/blob/279ea13ce25ef5b60e9b3a120d6a2add8e962f16/sos/report/plugins/kubernetes.py#L291-L292

Comment (arif-ali) [reply sos/report/plugins/networkmanager.py:121]: good catch, I think that would make sense, that would cover all files in the directory

File: sos/report/plugins/networkmanager.py [hunk 2]
```diff
@@ -110,16 +116,18 @@ def test_nm_status(version=1):
             "nmcli dev show .*": "nmcli_dev_show_sos"
         })
 
+    def postproc_system_connections(self, sc_path):
```
Comment (arif-ali) [inline sos/report/plugins/networkmanager.py:119]: curious question, the fact that we're doing the one thing, do we need a new function?

Comment (jcastill) [reply sos/report/plugins/networkmanager.py:119]: Probably now is not needed anymore. I'll fix this and resend.

**Review by arif-ali — APPROVED** [review]

**Pull request merged**